src/object-store/src/layers/lru_cache.rs

```diff
@@ -68,14 +68,14 @@ impl<I: Access> Layer<I> for LruCacheLayer {
 }
 
 #[derive(Debug)]
-pub struct LruCacheAccess<I> {
+pub struct LruCacheAccess<I, C> {
     inner: I,
-    read_cache: ReadCache,
+    read_cache: ReadCache<C>,
 }
 
-impl<I: Access> LayeredAccess for LruCacheAccess<I> {
+impl<I: Access, C: Access> LayeredAccess for LruCacheAccess<I, C> {
     type Inner = I;
-    type Reader = Arc<dyn ReadDyn>;
+    type Reader = Box<dyn ReadDyn>;
```

Comment:
The same question, why change it from `Arc` to `Box`?

Reply:
And we don't actually clone this point in the GreptimeDB codebase. It relies on the OpenDAL codebase to manage out this pointer. Perhaps we should use its exported type. Let me try.

Reply:
Done in c6e9f1293e25fcb55c1a87816ce7facda5e595e1